external-crates/move/tools/move-package/tests/test_sources/multiple_deps_from_lock_no_manifest/Move.toml

```diff
@@ -0,0 +1,19 @@
+# The purpose of this test is to make sure that the dependency graph is correctly read from lock
+# files (fast) even if the dependent manifest files have been stripped of their own dependency info
+# (which is of course still available in the respective lock files).
+#
+# The Move.locked file is not generated due to how the test harness works but the graph is correctly
+# included in the Move.resolved file. Please note that while dependency graph is correct, the
+# resolved graph is missing some information as it includes data from manifest files that are now
+# stripped.
```

Comment:
Okay, got it -- so this situation should never happen in practice, because we're tricking the resolution process by updating the manifest digests in C's and D's lock file?  In reality if we stripped the dependencies from their manifests, we would have regenerated their lock files as well and everything would be consistent.